catalogue_pipeline/transformer/src/test/scala/uk/ac/wellcome/transformer/SierraTransformerFeatureTest.scala

```diff
@@ -2,73 +2,87 @@ package uk.ac.wellcome.transformer
 
 import java.time.Instant
 
+import org.scalatest.concurrent.Eventually
 import org.scalatest.{FunSpec, Matchers}
 import uk.ac.wellcome.utils.JsonUtil._
 import uk.ac.wellcome.models.{
   IdentifierSchemes,
   SourceIdentifier,
-  UnidentifiedWork,
-  Work
-}
-import uk.ac.wellcome.transformer.utils.{
-  TransformableSQSMessageUtils,
-  TransformerFeatureTest
+  UnidentifiedWork
 }
+import uk.ac.wellcome.test.fixtures.{S3, SnsFixtures, SqsFixtures}
+import uk.ac.wellcome.test.utils.ExtendedPatience
+import uk.ac.wellcome.transformer.utils.TransformableMessageUtils
 import uk.ac.wellcome.utils.JsonUtil
 
 class SierraTransformerFeatureTest
     extends FunSpec
     with Matchers
-    with TransformerFeatureTest
-    with TransformableSQSMessageUtils {
+    with SqsFixtures
+    with SnsFixtures
+    with S3
+    with fixtures.Server
+    with Eventually
+    with ExtendedPatience
+    with TransformableMessageUtils {
 
-  override lazy val bucketName: String =
-    "test-sierra-transformer-feature-test-bucket"
-  val queueUrl: String = createQueueAndReturnUrl("test_sierra_transformer")
-  override val flags: Map[String, String] = Map(
-    "aws.region" -> "eu-west-1",
-    "aws.sqs.queue.url" -> queueUrl,
-    "aws.sqs.waitTime" -> "1",
-    "aws.sns.topic.arn" -> idMinterTopicArn,
-    "aws.metrics.namespace" -> "sierra-transformer",
-    "aws.s3.bucketName" -> bucketName
-  )
+  it("transforms sierra records and publishes the result to the given topic") {
+    withLocalSnsTopic { topicArn =>
+      withLocalSqsQueue { queueUrl =>
+        withLocalS3Bucket { bucketName =>
+          val flags: Map[String, String] = Map(
+            "aws.sqs.queue.url" -> queueUrl,
+            "aws.sns.topic.arn" -> topicArn,
+            "aws.s3.bucketName" -> bucketName,
+            "aws.sqs.waitTime" -> "1",
+            "aws.metrics.namespace" -> "sierra-transformer"
+          ) ++ s3LocalFlags ++ snsLocalFlags ++ sqsLocalFlags
 
-  it("should transform sierra records, and publish them to the given topic") {
+          withServer(flags) { _ =>
+            val id = "b001"
+            val title = "A pot of possums"
+            val lastModifiedDate = Instant.now()
 
-    val id = "b001"
-    val title = "A pot of possums"
-    val lastModifiedDate = Instant.now()
+            val sierraHybridRecordMessage =
```

Comment:
Move this out of the "with" block (I'll stop saying this now, but the general pattern applies).

Reply:
Generally why do you think this is a good idea?

I've decided I like it because it gives you the opportunity to define values as soon as possible so that non dependent processes can run concurrently (for example creating buckets/queues and starting the server).